NodeClassifier/utils/reader.py

```diff
@@ -23,36 +22,70 @@ def parse_packet_head(line):
     # Only generate a key if this packet contains IP information
     if len(data) < 2:
         return None
-    if data[2] != 'IP':
-        return None
+    # ipv4 packet
+    if data[2] == 'IP':
 
-    # Parse out the date and time the packet was seen
-    date_str = data[0] + ' ' + data[1]
-    date = datetime.datetime.strptime(date_str, '%Y-%m-%d %H:%M:%S.%f')
+        # Parse out the date and time the packet was seen
```

Comment:
a lot of this is repeated in both the `if` and the `elif` can you move it out so it's only in one place?